frontend/views/containers/user-settings/AppLogs.vue

```diff
@@ -0,0 +1,171 @@
+<template lang='pug'>
+  .settings-container
+    section.card
+      .c-header
+        fieldset.c-filters
+          legend.sr-only Visible logs
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='filter' value='error')
+            i18n Errors
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='filter' value='warn')
+            i18n Warnings
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='filter' value='debug')
+            i18n Debug
+
+        button.is-small.c-download(@click='downloadLogs')
+          i.icon-download.is-prefix.c-icon
+          i18n.hide-touch Download
+          i18n.hide-desktop Share
+        a(ref='linkDownload' hidden)
+
+      textarea.textarea.c-logs(ref='textarea' rows='12' readonly)
+        | {{ prettyLog }}
+
+      i18n.link(tag='a' @click='openTroubleshooting') Troubleshooting
+      br
+      br
+      p.has-text-danger DELETE BEFORE MERGE!
+      .buttons
+        button.is-small.is-outlined(@click='_createLog("error")') Log error
+        button.is-small.is-outlined(@click='_createLog("warn")') Log warn
+        button.is-small.is-outlined(@click='_createLog("debug")') Log debug
+        button.is-small.is-outlined(@click='_clearLogs') Clear Logs
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '@utils/events.js'
+import { downloadLogs, clearLogs } from '@model/captureLogs.js'
+export default {
+  name: 'AppLogs',
+  components: {},
+  data () {
+    return {
+      filter: [],
+      logs: []
+    }
+  },
+  mounted () {
+    this.filter = this.ourUserIdentityContract.settings.appLogsFilter
+    this.setLogs()
+    sbp('okTurtles.events/on', CAPTURED_LOGS, this.addLog)
+  },
+  beforeDestroy () {
+    sbp('okTurtles.events/off', CAPTURED_LOGS)
+  },
+  watch: {
+    async filter (value) {
+      sbp('okTurtles.events/emit', SET_APP_LOGS_FILTER, value)
+      await sbp('gi.actions/user/updateSettings',
+        { appLogsFilter: value },
+        this.$store.state.loggedIn.identityContractID
+      )
+    },
+    prettyLog () {
+      // Automatically scroll textarea to the bottom
+      this.$nextTick(() => {
+        if (this.$refs.textarea) {
+          this.$refs.textarea.scrollTop = this.$refs.textarea.scrollHeight
+        }
+      })
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'ourUserIdentityContract'
+    ]),
+    prettyLog () {
+      const isEntryNew = (type) => ['NEW_SESSION', 'NEW_VISIT'].includes(type)
+      return this.logs
+        .filter(({ type }) => {
+          return this.filter.includes(type) || isEntryNew(type)
+        })
+        .map(({ type, msg }) => {
+          return isEntryNew(type)
+            ? `--------------- \n [${type}] :: ${msg}`
+            : `[${type}] :: ${msg.map(something => JSON.stringify(something)).join(' ')}`
```

Comment:
Avoid the use of `::`, just have it nice and compact, e.g. `<Timestamp> [debug] my great log message` (where `<Timestamp>` is the ISOString e.g. `new Date().toISOString()`, which will need to be stored with the messages as well)